src/com/google/enterprise/adaptor/database/ResponseGenerator.java

```diff
@@ -386,19 +395,80 @@ public void generateResponse(ResultSet rs, Response resp)
         throws IOException, SQLException {
       overrideContentType(rs, resp);
       overrideDisplayUrl(rs, resp);
-      Blob blob = rs.getBlob(getContentColumnName());
-      InputStream in = blob.getBinaryStream();
-      OutputStream out = resp.getOutputStream();
-      com.google.enterprise.adaptor.IOHelper.copyStream(in, out);
-      in.close();
-      if (!(blob instanceof javax.sql.rowset.serial.SerialBlob)) {
-        // SerialBlob is adamant about not supporting free
-        try {
-          blob.free();
-        } catch (java.sql.SQLFeatureNotSupportedException | 
-            UnsupportedOperationException unsupported) {
-          // let JVM garbage collection deal with it
+
+      ResultSetMetaData rsMetaData = rs.getMetaData();
+      int columnType = Types.OTHER;
```

Comment:
Since this is what you want, putting it second among these declarations makes no sense. I would typically pick first.